homeassistant/components/screenlogic/binary_sensor.py

```diff
@@ -12,16 +17,61 @@
 
 from . import ScreenlogicDataUpdateCoordinator
 from .const import DOMAIN
-from .entity import ScreenlogicEntity, ScreenLogicPushEntity
+from .data import EntityParameter, SupportedDeviceDescriptions, process_supported_values
+from .entity import (
+    ScreenlogicEntity,
+    ScreenLogicEntityDescription,
+    ScreenLogicPushEntity,
+    ScreenLogicPushEntityDescription,
+)
 
-SL_DEVICE_TYPE_TO_HA_DEVICE_CLASS = {DEVICE_TYPE.ALARM: BinarySensorDeviceClass.PROBLEM}
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_DATA: SupportedDeviceDescriptions = {
+    DEVICE.CONTROLLER: {
+        GROUP.SENSOR: {
+            VALUE.ACTIVE_ALERT: {},
+            VALUE.CLEANER_DELAY: {},
+            VALUE.FREEZE_MODE: {},
+            VALUE.POOL_DELAY: {},
+            VALUE.SPA_DELAY: {},
+        },
+    },
+    DEVICE.PUMP: {
+        "*": {
+            VALUE.STATE: {},
+        },
+    },
+    DEVICE.INTELLICHEM: {
+        GROUP.ALARM: {
+            VALUE.FLOW_ALARM: {},
+            VALUE.ORP_HIGH_ALARM: {},
+            VALUE.ORP_LOW_ALARM: {},
+            VALUE.ORP_SUPPLY_ALARM: {},
+            VALUE.PH_HIGH_ALARM: {},
+            VALUE.PH_LOW_ALARM: {},
+            VALUE.PH_SUPPLY_ALARM: {},
+            VALUE.PROBE_FAULT_ALARM: {},
+        },
+        GROUP.ALERT: {
+            VALUE.ORP_LIMIT: {},
+            VALUE.PH_LIMIT: {},
+            VALUE.PH_LOCKOUT: {},
+        },
+        GROUP.WATER_BALANCE: {
+            VALUE.CORROSIVE: {},
+            VALUE.SCALING: {},
+        },
+    },
+    DEVICE.SCG: {
+        GROUP.SENSOR: {
+            VALUE.STATE: {},
+        },
+    },
+}
```

Comment:
Is there any HA specific logic in these blocks? Could they be moved into the library instead so we don't have to maintain it here?

Reply:
Not currently. I had a few goals in mind with this arrangement:
1. Decouple HA's exposed data points from the library's exposed data points, and have HA's data points under the control of HA, and not the library. It doesn't feel appropriate to me to have a 3rd party library be so potentially in control of the HA user's experience.
3. Maintain continuity with the `sensor` data structure, which required specific conditions for specific sensors. I felt continuity would help any potential future maintainers more easily understand where the data is coming from.
5. Future-proof the definition of supported data points across the integration with a data structure that could work for everything. It's very possible that additional data is exposed in the future that will use this ability.

That said, if this goes against HA development principles or I'm thinking about it in the wrong way, I would absolutely appreciate any guidance in this aspect.

Reply:
There isn't anything wrong with keeping it here from an architecture perspective.  

Mostly I'm hoping for less code to review/maintain on the HA side but it looks like this mapping is effectively library to HA mapping so its in the right place.